### PR TITLE
Adds width and flags '-', '^', '_', and '0' to date filter when using strftime.

### LINKED ISSUE #134: Day of month (%-d) does nothing when zero-padding and Ruby syntax
When you normally format a date using Ruby syntax you do something like:

{{ myDate | date: "%m %d, %Y" }}
If the myDate is January 12, 2014, then I'd get
01 12, 2014

Fine.

If myDate is:
January 4, 2014, then I'd get:
01 04, 2014

But if I want:
01 **4**, 2014
I should be able to have:
{{ myDate | date: "%m %-d, %Y" }}
But that returns:
01 %-d, 2014

### PR DESCRIPTION
**Pull request by mikebridge**
closes #134

### CODE REVIEW

**Comment by mikebridge** [comment]
See: https://github.com/dotliquid/dotliquid/pull/138/files

**Comment by haf** [comment]
Hi @mikebridge could you please rebase this on master so we can get it merged?

**Comment by mikebridge** [comment]
Ok, this is updated.

I'm glad to see this project is finally getting some attention!  I finally ended up creating my [own port](https://github.com/mikebridge/Liquid.NET) of Liquid from scratch because this one had been abandoned for so many years.

File: src/DotLiquid.Tests/StandardFilterTests.cs [hunk 1]
```diff
@@ -1,281 +1,287 @@
...
-        }
-
-        [Test]
-        public void TestModulo()
-        {
-            Helper.AssertTemplateResult("1", "{{ 3 | modulo:2 }}");
-        }
-    }
-}
+using System;
```
Comment (haf) [inline src/DotLiquid.Tests/StandardFilterTests.cs]: I think you've replaced CRLF with LF here.

**Comment by haf** [comment]
Hi Mike. Thanks. Perhaps you could port the bugfixes and (if needed) parser?

**Comment by haf** [comment]
Closing since there was no answer. Can be re-opened.

**Comment by mikebridge** [comment]
Yes, if I get a minute I'll fix the line endings.

**Comment by laedit** [comment]
Reopen to see if it can be merged.

**Pull request closed without merging**